### PR TITLE
[Zero-Dim] support input 0D Tensor as scalar attribute for some api

### PR DESCRIPTION
**Pull request by zhwesky2010**
### PR types

New features
### PR changes

APIs
### Describe

0D Tensor作为标量属性，例如API的shape、start、end时，包含：
---

```
paddle.slice
paddle.strided_slice
paddle.linspace
paddle.arange
paddle.normal
paddle.rand
paddle.randn
paddle.randint
paddle.randint_like
paddle.standard_normal
paddle.uniform
paddle.empty
paddle.empty_like
paddle.full
paddle.full_like
paddle.ones
paddle.ones_like
paddle.zeros
paddle.zeros_like
```

### CODE REVIEW

**Review by jeff41404 — DISMISSED** [review]
LGTM

File: python/paddle/tensor/creation.py [hunk 1]
```diff
@@ -690,7 +696,9 @@ def zeros(shape, dtype=None, name=None):
     Creates a tensor of specified :attr:`shape` and :attr:`dtype`, and fills it with 0.
 
     Args:
-        shape(tuple|list|Tensor): Shape of the Tensor to be created, the data type of ``shape`` is int32 or int64.
+        shape (tuple|list|Tensor): Shape of the Tensor to be created. The data type is ``int32`` or ``int64`` .
+            If ``shape`` is a list or tuple, each element of it should be integer or OD Tensor with shape [].
```
Comment (Ligoml) [inline python/paddle/tensor/creation.py]:
```suggestion
            If ``shape`` is a list or tuple, each element of it should be integer or 0-D Tensor with shape [].
```

File: python/paddle/tensor/creation.py [hunk 2]
```diff
@@ -620,7 +620,9 @@ def ones(shape, dtype=None, name=None):
     Create a Tensor of specified :attr:`shape` and :attr:`dtype` and fill it with 1.
 
     Args:
-        shape (tuple|list|Tensor): Shape of the Tensor to be created, the data type of shape should be int32 or int64.
+        shape (tuple|list|Tensor): Shape of the Tensor to be created. The data type is ``int32`` or ``int64`` .
+            If ``shape`` is a list or tuple, the elements of it should be integers or OD Tensor with shape [].
```
Comment (Ligoml) [inline python/paddle/tensor/creation.py]:
```suggestion
            If ``shape`` is a list or tuple, the elements of it should be integers or 0-D Tensor with shape [].
```

Comment (zhwesky2010) [reply python/paddle/tensor/creation.py]: Fixed, thx

File: python/paddle/tensor/creation.py (hunk 1 above)
Comment (zhwesky2010) [reply python/paddle/tensor/creation.py]: Fixed, thx

**Review by Ligoml — APPROVED** [review]
LGTM for docs

**Review by luotao1 — APPROVED** [review]
LGTM for PADDLE_ENFORCE

**Review by jeff41404 — APPROVED** [review]

**Pull request merged**